src/coverage.ts

```diff
@@ -81,22 +99,36 @@ export const coverage = async ({
   })
 
   const { linesHit, linesFound } = diffCoverage
-  const { data: diffCoverageStatus } = await createCommitStatus({
-    token: utoken ? utoken : ghToken,
-    context,
-    commitContext: 'codelyze/patch',
-    state: 'success',
-    description:
-      linesFound > 0
-        ? `${percentString(linesHit / linesFound)} of diff hit`
-        : 'No diff detected'
-  })
+  const diffCoverageRate = linesHit / linesFound
+  console.log(
+    `diffCoverage: ${diffCoverageRate} | differenceThreshold: ${differenceThreshold}`
+  )
+  let diffCoverageStatus
+  if (!(emptyPatch && linesFound === 0)) {
+    const { data } = await createCommitStatus({
+      token: utoken ? utoken : ghToken,
+      context,
+      commitContext: 'codelyze/patch',
+      state: evaluateState([diffCoverageRate * 100 >= patchThreshold]),
+      description:
+        linesFound > 0
+          ? `${percentString(linesHit / linesFound)} of diff hit`
+          : 'No diff detected'
+    })
+    diffCoverageStatus = data
+  }
 
-  addAnnotations(diffCoverage.uncoveredHunks)
+  if (shouldAddAnnotation) {
+    addAnnotations(diffCoverage.uncoveredHunks)
+  }
 
   return { status, rate, diff, diffCoverageStatus }
 }
 
+export const evaluateState = (conditions: boolean[]) => {
+  return conditions.every(Boolean) ? 'success' : 'failure'
+}
```

Comment:
You should always use booleans to signify success or failure, and only convert to the string representation where needed. Also, do you really need to export it?

```suggestion
const evaluateState = (conditions: boolean[]) => conditions.every(Boolean)
```

Reply:
applied